File: rperz/QSLib/QSLib/AST/Expressions/Unary/Not.cs
using QSLib.AST.Types;
using QSLib.Visitors;
namespace QSLib.AST.Expressions.Unary
{
    public class Not : Unary_Expression
    {

        public Not(QSExpression left, int lineNr)
            : base(left, lineNr)
        {

        }


        #region TypeChecker
        public override T Accept<T>(IExpressionVisitor<T> checker)
        {
            return checker.Visit(this);
        }
        #endregion

        #region Getters
        public override string GetOperator()
        {
            return "!";
        }

        public override QSType GetType(TypeMemory memory)
        {
            return new BoolType();
        }

        #endregion

    }
}

File: eenass/QLanguage/src/ast/expr/unExpr/Not.java
package ast.expr.unExpr;

import typecheck.SymbolTable;
import ast.expr.Expr;
import ast.types.BoolType;
import ast.types.Type;
import ast.visitors.ExprVisitor;

public class Not extends UnExpr{
	
	public Not(Expr operand){
		super(operand);		
	}

	@Override
	public <T> T accept(ExprVisitor<T> visitor) {
		return visitor.visit(this);
	}

	@Override
	public Type typeof(SymbolTable symbolTable) {
		return new BoolType();
	}
}
